src/main/java/org/folio/edge/rtac/RtacHandler.java

```diff
@@ -39,76 +41,207 @@
+              badRequest(ctx, "Invalid instance id" + params.toString());
+              return;
+            }
+
+            var rtacParams = new HashMap<>();
+            rtacParams.put(PARAM_INSTANCE_IDS, ids);
+            rtacParams
+                .put(PARAM_FULL_PERIODICALS, Boolean.valueOf(params.get(PARAM_FULL_PERIODICALS)));
+            instanceIds = Mappers.jsonMapper.writeValueAsString(rtacParams);
+          } catch (JsonProcessingException e) {
+            logger.error("Exception during serialization in mod-rtac", e);
+            returnEmptyResponse(ctx);
             return;
           }
 
-          var rtacParams = new HashMap<>();
-          rtacParams.put(PARAM_INSTANCE_IDS, ids);
-          rtacParams.put(PARAM_FULL_PERIODICALS, Boolean.valueOf(params.get(PARAM_FULL_PERIODICALS)));
-          instanceIds = Mappers.jsonMapper.writeValueAsString(rtacParams);
-        } catch (JsonProcessingException e) {
-          logger.error("Exception during serialization in mod-rtac", e);
-          returnEmptyResponse(ctx);
-          return;
-        }
-        rtacClient.rtac(instanceIds, request.headers())
-          .thenAcceptAsync(body -> {
-            try {
-              logger.debug("rtac response: {}", body);
-              String xml;
-              final var instances = Instances.fromJson(body);
-              if (isBatch) {
-                xml = instances.toXml();
-              } else {
-                var holdings = instances.getHoldings().isEmpty() ? new Holdings() : instances.getHoldings().get(0);
-                xml = holdings.toXml();
-              }
-              logger.info("Converted Response: \n {}", xml);
-              ctx.response()
-                .setStatusCode(200)
-                .putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_XML)
-                .end(xml);
-            } catch (IOException e) {
-              logger.error("Exception translating JSON -> XML: {}", e.getMessage(), e);
-              returnEmptyResponse(ctx);
+          final boolean isAcceptHeaderExists = checkAcceptHeadersExists(request);
+          // Remember original Accept headers
+          final List<String> originAcceptHeaders = request.headers().getAll(ACCEPT);
+          final boolean acceptHeadersUpdateFlag;
+
+          if (!isAcceptHeaderExists) {
+            // Provide "json" accept header for further request to RTAC system
+            updateAcceptHeadersWith(request, Arrays.asList(JSON_MIME_TYPE));
+            acceptHeadersUpdateFlag = true;
+          } else {
+            // Replace xml-related accept header by JSON ones, to make a call to RTAC (it supports JSON by default)
+            if (checkAcceptHeadersByPattern(request, XML_RELATED_MIME_TYPE_PATTERN, false)) {
+              updateAcceptHeadersByPattern(request, XML_RELATED_MIME_TYPE_PATTERN, JSON_MIME_TYPE);
+              acceptHeadersUpdateFlag = true;
+            } else {
+              acceptHeadersUpdateFlag = false;
             }
```

Comment:
If the subsequent requests only ever requests JSON, why not always use that accept header irrespective of what the client asks this module to produce?

Reply:
That make a sense, but to workaround only exceptional cases related to XML/JSON I was replacing "Accept" headers for JSON. This allows to pass rest of header's values processing further by chain. But to be honest, here I have some doubts as for case with masking any Accept header values with JSON with further restoring them to the client. Possibly finally it would be only one right approach. I slightly rewrote it.

Reply:
Thanks for reworking it. 

I guess I don't know why we are passing on all of the headers, considering we are taking an external request (which has no token or tenant) and making an internal request.

Maybe @mreno-EBSCO or @craigmcnally can help me understand. Is it common practice for the edge modules to use the headers from the incoming request for the internal request(s)?

I know we do this in the business logic modules, however those are internal (from Okapi) to internal (back to Okapi) generally.

Reply:
The main concern here is the way of passing particular header we deal with, I mean "Accept".  API works with headers in a way that suppose usage of several "Accept" (or any other headers) passing, I mean request.getHeaders().getAll(<header_name>) returns list of headers. So we can make a conclusion that client can pass several headers at once, like: 
- Accept: application/xml;
- Accept: application/json 
etc.
From other side, headers can be passed in single entry, but separated by coma, like:
- Accept: application/xml, application/json
etc.
So, to conclude all these, and taking into account that internaly we use only json payload - we really can replace all the Accept headers by JSON and then restore them after receiving payload from RTAC subsystem. That of course can avoid extra processing and make code less error prone.
Does it make sense? I mean just substitute request type temporary.

Reply:
@marcjohnson-kint, I am not aware of any headers that the edge modules would need to forward to a FOLIO module. Maybe tracking headers? It looks like this could be used to override the token/tenant retrieved via login, which seems odd, but maybe there is a use case during testing. The edge modules I have written typically use `null` for the client headers when calling FOLIO modules, so only the default headers are added.

Reply:
> I am not aware of any headers that the edge modules would need to forward to a FOLIO module.

Thanks

> It looks like this could be used to override the token/tenant retrieved via login, which seems odd, but maybe there is a use case during testing. 

This seems like unnecessary complexity to me and a potential security issue.

I would prefer a simpler code base to maintain where the internal request is constructed as independently of the external request as possible.

This may well be considered out of scope for this change.

Reply:
I think it is in scope here since we are adding some convoluted code to preserve the Accept header. Unless @craigmcnally can think of a reason to pass on the client side headers, I'd say let's remove this capability and simplify the code.

Reply:
Got it, thank you, will simplify the code.